dlio_benchmark/reader/reader_factory.py

```diff
@@ -43,18 +43,32 @@ def get_reader(type, dataset_type, thread_index, epoch_number):
         elif type == FormatType.CSV:
             from dlio_benchmark.reader.csv_reader import CSVReader
             return CSVReader(dataset_type, thread_index, epoch_number)
-        elif type == FormatType.JPEG:
-            from dlio_benchmark.reader.jpeg_reader import JPEGReader
-            return JPEGReader(dataset_type, thread_index, epoch_number)
-        elif type == FormatType.PNG:
-            from dlio_benchmark.reader.png_reader import PNGReader
-            return PNGReader(dataset_type, thread_index, epoch_number)
+        elif type == FormatType.JPEG or type == FormatType.PNG:
```

Comment:
This is very verbose, I think for this class, we need to upgrade from a factory pattern to a builder pattern. 

The idea is to 
Switch the reader to a builder name and then do something like.
```
Builder.loader(DALI).format(NPY).construct().
```
This article conceptually explains it https://www.baeldung.com/creational-design-patterns

Reply:
This can be done in future to improve it.

Reply:
Still need to spend sometime to understand it and implemented it. 

Is it OK if you try to implement it and send the PR?

Reply:
I think for now it’s fine. I can come back and fix it.